projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,281 @@
+  private boolean sentHaser = true;
+  private boolean sentMoreThanFiveLawsuits = false; // The past year
+  private Date receivedSpamDate = new Date();
+
+  // region Lawsuit form fields
+  // General
+  private String spamType = "הודעה אלקטרונית";
+  private String claimAmount = "סכום תביעה";
+  private String moreThanFiveLawsuits = "הגיש";
+  private String lessThanFiveLawsuits = "לא הגיש/ו";
+  private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+  private String claimCaseSubscription =
+      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+
+  // User
+  private String userPrivateName = "שם פרטי";
+  private String userLastName = "שם משפחה";
+  private String userID = "ת ז משתמש";
+  private String userAddress = "כתובת משתמש";
+  private String userPhone = "טלפון משתמש";
+  private String userFax = "פקס משתמש";
+
+  // First spam company's details
+  private String companyName = "שם החברה";
+  private String companyId = "מספר ח.פ";
+  private String companyAddress = "כתובת החברה";
+  private String companyPhone = "טלפון החברה";
+  private String companyFax = "פקס חברה";
+
+  // Second spam company's details
+  private String company2Name = "שם החברה 2";
+  private String company2Id = "מספר ח.פ 2";
+  private String company2Address = "כתובת החברה 2";
+  private String company2Phone = "טלפון החברה 2";
+  private String company2Fax = "פקס חברה 2";
+  // endregion
+
+  Button createPdfButton;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_lawsuit_pdf);
+    createPdfButton = (Button) findViewById(R.id.button_createPDF);
+
+    initDirStructure();
+    setTimeZones();
+
+    createPdfButton.setOnClickListener((View v) -> checkPermissionsThenCreatePdf());
+  }
+
+  private void createPdf() {
+    // TODO: Success reading license file only from "res/raw". Perhaps should be read from different
+    // dir
+    // Activate iText license
+    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+    try {
+      HtmlConverter.convertToPdf(
+          fillTemplate(),
+          new FileOutputStream(Paths.get(lawsuitOutputPath, getPdfFilename()).toString()));
+      Toast.makeText(LawsuitPdfActivity.this, "Lawsuit created!", Toast.LENGTH_LONG).show();
+    } catch (IOException e) {
+      Log.w(TAG, "Error creating PDF Lawsuit");
+      e.printStackTrace();
+    }
+  }
+
+  private String fillTemplate() {
+
+    String lawsuit = null;
+    try {
+      lawsuit =
+          new String(
+              Files.readAllBytes(Paths.get(lawsuitTemplateFilePath)), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      Log.w(TAG, "Error reading lawsuit's template.");
+      e.printStackTrace();
+    }
+
+    // General form fields
+    lawsuit =
+        lawsuit.replace("&lt;claimCase&gt;", sentHaser ? claimCaseHaser : claimCaseSubscription);
```

Comment:
You'll still get an NPE here. You can try it and see. Pass a bad `lawsuitTemplateFilePath`.

Reply:
Fixed it as to our last conversation.

Write output / Read template exceptions are handled on `createPdf()`.